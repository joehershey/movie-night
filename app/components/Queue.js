import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
  ScrollView,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

import Movie from "../components/Movie";
import Filters from "../components/Filters";

function Queue(props) {
  const [page, setPage] = useState(1);
  const [watchProviders, setWatchProviders] = useState([]);
  const [genres, setGenres] = useState([]);
  var movies = [];
  const [htmlMovies, setHtmlMovies] = useState([]);

  React.useEffect(() => {
    const unsubscribe = props.navigation.addListener("focus", () => {
      getMovies();
    });

    return unsubscribe;
  }, [props.navigation]);

  function getMovies(loadMore = false) {
    fetch(props.url + "group/" + props.group_id + "/movies/" + props.user_id, {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json; charset=utf-8",
      },
      method: "GET",
    })
      .then((response) => response.json())
      .then((responseJson) => {
        movies = [];
        console.log(responseJson);
        let test = [
          { tmdb_movie_id: 566525, group_rating: 4.7, user_rating: 5 },
          { tmdb_movie_id: 438631, group_rating: 9.2, user_rating: 5 },
          { tmdb_movie_id: 580489, group_rating: 7.3, user_rating: 5 },
          { tmdb_movie_id: 574060, group_rating: 1.3, user_rating: 5 },
          { tmdb_movie_id: 630004, group_rating: 3.3, user_rating: 5 },
        ];
        for (const [i, movie] of test.entries()) {
          movies.push(movie);
          console.log(movie);
          addMovieData(movie, i, i == test.length - 1);
        }
      })
      .catch((error) => {
        console.error(error);
      });
  }

  const BASEURL = "https://api.themoviedb.org/3/";
  const APIKEY = "?api_key=3eb4bede8c2782fba9b6b7cd9c56b62c";

  function addMovieData(movie, idx, set = false) {
    let TEST = BASEURL + "movie/" + movie.tmdb_movie_id + APIKEY;
    //let TEST = BASEURL + "movie/" + 72 + APIKEY;
    fetch(TEST, {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json; charset=utf-8",
      },
      method: "GET",
    })
      .then((response) => response.json())
      .then((responseJson) => {
        movies[idx].data = responseJson;
        console.log(movies[idx]);
        if (!set) return;
        movies.sort((a, b) => (a.group_rating < b.group_rating ? 1 : -1));
        setHtmlMovies(movies);
      })
      .catch((error) => {
        console.error(error);
      });
  }

  function setRatingAPI(rating) {
    console.log("*****" + rating + "*****");
  }

  const moviesToRender = [];
  for (const [i, movie] of htmlMovies.entries()) {
    console.log(movie.data);
    moviesToRender.push(
      /* Collapsable card with member info/edit if admin */
      <Movie
        key={i}
        url={props.url}
        user_id={props.user_id}
        group_id={props.group_id}
        movie={movie.data}
        group_rating={movie.group_rating}
        user_rating={movie.user_rating}
        setRatingAPI={(rtg) => setRatingAPI(rtg)}
      ></Movie>
    );
  }

  return (
    <View style={{ flex: 11 }}>
      <ScrollView
        style={{
          flexDirection: "column",
        }}
        contentContainerStyle={{
          justifyContent: "flex-start",
          alignItems: "stretch",
        }}
      >
        {/* <Filters
          genres={genres}
          watchProviders={watchProviders}
          setGenres={() => setGenres}
          setWatchProviders={() => setWatchProviders}
          getMovies={() => getMovies()}
          setPage={(a) => console.log("N/A")}
        ></Filters> */}

        {moviesToRender}

        <View style={{ height: 50 }}></View>
      </ScrollView>
    </View>
  );
}

export default Queue;
