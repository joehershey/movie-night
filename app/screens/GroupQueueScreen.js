import React from "react";
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
  SafeAreaView,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

import TabBar from "../components/TabBar";
import TopBar from "../components/TopBar";
import Queue from "../components/Queue";

import fetch from "cross-fetch";

function GroupQueueScreen(props) {
  const [isAdmin, setIsAdmin] = React.useState(false);

  React.useEffect(() => {
    const unsubscribe = props.navigation.addListener("focus", () => {
      getUsersAPI();
    });

    return unsubscribe;
  }, [props.navigation]);

  //used to see if the user is an admin
  function getUsersAPI() {
    fetch(props.url + "group/" + props.group_id + "/users", {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json; charset=utf-8",
        Authorization: "Bearer " + props.token,
      },
      method: "GET",
    })
      .then((response) => response.json())
      .then((responseJson) => {
        members = responseJson;
        for (let i = 0; i < members.length; i++) {
          if (members[i].user_id == props.user_id) {
            setIsAdmin(members[i].is_admin ? true : false); //member isAdmin set true if active user is_admin
            break;
          }
        }
      })
      .catch((error) => {
        console.error(error);
      });
  }

  console.log("on queue");
  return (
    <SafeAreaView style={STYLES.container}>
      {/* Top Bar */}
      <TopBar
        navigation={props.navigation}
        screen="Landing"
        message="Groups"
      ></TopBar>
      {/* Content */}
      <View style={[STYLES.content, { backgroundColor: COLORS.primary }]}>
        <Queue
          navigation={props.navigation}
          url={props.url}
          user_id={props.user_id}
          group_id={props.group_id}
          token={props.token}
          isAdmin={isAdmin}
        ></Queue>
      </View>
      {/* Tabs */}
      <TabBar navigation={props.navigation} screen="queue"></TabBar>
    </SafeAreaView>
  );
}

export default GroupQueueScreen;
