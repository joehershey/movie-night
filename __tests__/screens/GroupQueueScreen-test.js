import GroupQueueScreen from '../../app/screens/GroupQueueScreen';
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { TouchableWithoutFeedback } from "react-native";
import { shallow, configure } from 'enzyme';
import Queue from "../../app/components/Queue";


configure({adapter: new Adapter()});

describe('Test Group Queue Screen', () => {
    beforeEach( async () => {
        addListener = jest.fn();
        props = {
            navigation: {
                addListener
            }
        }
        global.tree = renderer.create(<GroupQueueScreen {...props}/>).toJSON();
    });

    it("Group queue screen renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });
});