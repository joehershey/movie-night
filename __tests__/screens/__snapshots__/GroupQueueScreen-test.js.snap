// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Group Queue Screen Group queue screen renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        Groups
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "paddingBottom": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginBottom": 0,
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 11,
        },
        Object {
          "backgroundColor": "#1d1f1e",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 11,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "stretch",
            "justifyContent": "flex-start",
          }
        }
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "height": 50,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Schedule
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#166fdb",
            "fontWeight": "bold",
            "margin": 5,
          }
        }
      >
        Queue
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Search
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Settings
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
